### PR TITLE
207 mixed add gender to body diagram dynamically resize the body

### PR DESCRIPTION
**Pull request by AaronFmyHub**
# What type of PR is this? (Check all that apply)

- [x] ✨**feature**: Introduces completely new code or new features.
- [x] 🐛**fix**: Implements changes that fix a bug. Ideally, reference an issue if present.
- [ ] ♻️**refactor**: Includes any code-related change that is neither a fix nor a feature.
- [ ] ✅**build**: Encompasses all changes related to the build of the software, including changes to dependencies or the addition of new ones.
- [x] ⚡️**test**: Pertains to all changes regarding tests, whether adding new tests or modifying existing ones.
- [ ] 🚰**ci**: Involves all changes related to the configuration of continuous integration, such as GitHub Actions or other CI systems.
- [ ] 📚**docs**: Includes all changes to documentation, such as README files, or any other documentation present in the repository.
- [ ] 🗑️**chore**: Captures all changes to the repository that do not fit into the above categories.

# Description

Added gender to user profile and used that as the default for displaying body model. Also dynamically resize the SVG ViewBox using user device specifications.
1. Add gender to schema and global context following current method.
2. Create toggle that is non-indicative.
3. Use only screen height to limit size of the SVG ViewBox since the ViewBox we are using is always 1-to-2 in terms of width and height. Also protect the bottom part, which contains functional buttons, from being overlapped on.

# Tests

How was this tested?

- [x] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [x] Manual tests
- [ ] Tests were NOT needed
- [ ] Other (explain below)

# [Optional] Screenshots
[genderSelection.webm](https://github.com/UNLV-CS472-672/2024-S-GROUP3-Barbell/assets/146596967/60ed364f-823e-40d6-bb98-377beafa433d)

image
User has PREFERNOTSAY as gender, so use default.

# Documentation

- [ ] Added to README.me
- [ ] Seperate document
- [x] NO documentation needed

# AI Usage:

Used AI to simplify if-else statement, AI improved run time and readability of my code.

### CODE REVIEW

File: apps/expo/src/components/muscleGroup/GenderSwitch.test.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import { render } from '@testing-library/react-native'
+import Toggle from '~/components/toggle/toggle'
```
Comment (thienguen) [inline apps/expo/src/components/muscleGroup/GenderSwitch.test.tsx:3]: remove unused import

File: apps/expo/src/components/muscleGroup/GenderSwitch.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useRef, useState } from 'react'
```
Comment (thienguen) [inline apps/expo/src/components/muscleGroup/GenderSwitch.tsx:1]: also run prettier on this one

File: apps/expo/src/components/toggle/Toggle.test.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import { render } from '@testing-library/react-native'
+import Toggle from '~/components/toggle/toggle'
```
Comment (thienguen) [inline apps/expo/src/components/toggle/Toggle.test.tsx:3]: please remove this test file...there is already one that more rigid written for you under __tests__ folder

image

File: apps/expo/src/components/muscleGroup/GenderSwitch.test.tsx (hunk 1 above)
Comment (thienguen) [reply apps/expo/src/components/muscleGroup/GenderSwitch.test.tsx:3]: also, move this file under __tests__ folder

**Review by thienguen — CHANGES_REQUESTED** [review]
(Optional): use tailwindcss, but since we short on time, if you can't do it immediately, we can move on

**Comment by thienguen** [comment]
image

good works, run on my machine locally. One thing I do feel iffy is that we currently do not have the pop-up after sign-in asking users what gender they are, this can be a separate issue.

Also, try to sync your branch to the latest main, a lot has been changed,

File: apps/expo/src/app/muscleGroup/index.tsx [hunk 4]
```diff
@@ -1,14 +1,17 @@
 import type { BodyPart } from 'react-native-body-highlighter'
+
```
[the remark this answers is not in the document]
Comment (AaronFmyHub) [reply apps/expo/src/app/muscleGroup/index.tsx:2]: I'll remember that

File: apps/expo/src/components/muscleGroup/GenderSwitch.tsx (hunk 2 above)
Comment (AaronFmyHub) [reply apps/expo/src/components/muscleGroup/GenderSwitch.tsx:1]: Thanks

File: apps/expo/src/components/muscleGroup/GenderSwitch.test.tsx (hunk 1 above)
Comment (AaronFmyHub) [reply apps/expo/src/components/muscleGroup/GenderSwitch.test.tsx:3]: Moved and removed

File: apps/expo/src/components/toggle/Toggle.test.tsx (hunk 3 above)
Comment (AaronFmyHub) [reply apps/expo/src/components/toggle/Toggle.test.tsx:3]: Removed

**Comment by thienguen** [comment]
thank you so much! for implementing the changes right after my review

**Review by thienguen — APPROVED** [review]

**Pull request merged**